server/src/main/java/org/elasticsearch/index/mapper/BlockDocValuesReader.java

```diff
@@ -504,6 +506,80 @@ public String toString() {
         }
     }
 
+    public static class DenseVectorBlockLoader extends DocValuesBlockLoader {
```

Comment:
Added a BlockLoader for dense vectors, that uses `FloatVectorValues` to retrieve indexed vector data.